python/modelUtils.py

```diff
@@ -174,25 +174,29 @@ def get_model(input_shape, nclass=2, model_name='dense_100x3'):
 
 def train_model(model, X, Y, w, callbacks=[], figname='', batch_size=32768, epochs=100, verbose=1, model_filepath=None):
 
-    # initalize empty lists
-    X_train_list, X_val_list, Yw_train_list, Yw_val_list = [], [], [], []
+    # initalize empty dictionaries
+    train_dictionary, val_dictionary = {}, {}
+    train_y_dictionary, val_y_dictionary = {}, {}
+
+    train_w, val_w = [], []
+
+    X_train, X_val, Y_train, Y_val = train_test_split(X, Y, random_state=325)
 
-    # prepare the lists
+    # prepare the dictionaries
     for i in range(n_models_in_parallel):
-        X_train, X_val, Y_train, Y_val, w_train, w_val = train_test_split(X, Y, w[i])
+        w_train, w_val = train_test_split(w[i], random_state=325)
```

Comment:
The purpose of using an integer for the random_state is to ensure the train test splitting is the same for data arrays and for all weight arrays right? Instead of always using a hard coded magic number 325, would it be possible to get a random number every time train model is called and use it as the random_state?

Reply:
Done. Replaced it with a dynamically generated random state that changes each time train_model is called but the same across each parallel models.